arrow-core-test/src/main/kotlin/arrow/test/laws/FoldableLaws.kt

```diff
@@ -35,44 +44,47 @@ object FoldableLaws {
     val GEN: Gen<Kind<F, Int>> = GENK.genK(Gen.intSmall())
+    val GENListK: Gen<Kind<F, ListK<Int>>> = GENK.genK(Gen.listK(Gen.intSmall()))
     val GENB = GENK.genK(Gen.bool())
 
-    val EQ = Int.eq()
+    val EQ: Eq<Int> = Int.eq()
     val EQOptionInt = Option.eq(Int.eq())
-
-    return listOf(
-      Law("Foldable Laws: foldRight is lazy") { FF.`foldRight is lazy`(GEN, EQ) },
-      Law("Foldable Laws: Left fold consistent with foldMap") { FF.leftFoldConsistentWithFoldMap(GEN, EQ) },
-      Law("Foldable Laws: Right fold consistent with foldMap") { FF.rightFoldConsistentWithFoldMap(GEN, EQ) },
-      Law("Foldable Laws: find matching predicate should return some(value) or none") { FF.`find matching predicate should return some(value) or none`(GEN, EQOptionInt) },
-      Law("Foldable Laws: Exists is consistent with find") { FF.existsConsistentWithFind(GEN) },
-      Law("Foldable Laws: Exists is lazy") { FF.existsIsLazy(GEN, EQ) },
-      Law("Foldable Laws: ForAll is lazy") { FF.forAllIsLazy(GEN, EQ) },
-      Law("Foldable Laws: ForAll consistent with exists") { FF.forallConsistentWithExists(GEN) },
-      Law("Foldable Laws: ForAll returns true if isEmpty") { FF.forallReturnsTrueIfEmpty(GEN) },
-      Law("Foldable Laws: FoldM for Id is equivalent to fold left") { FF.foldMIdIsFoldL(GEN, EQ) },
-      Law("Foldable Laws: firstOrNone returns None if isEmpty") { FF.firstOrNoneReturnsNoneIfEmpty(GEN) },
-      Law("Foldable Laws: firstOrNone returns None if predicate fails") { FF.firstOrNoneReturnsNoneIfPredicateFails(GEN) },
-      Law("Foldable Laws: firstOrNone is consistent with find") { FF.`firstOrNone is consistent with find`(GEN, EQOptionInt) },
-      Law("Foldable Laws: firstOrNone is consistent with find matching predicate") { FF.`firstOrNone is consistent with find predicate`(GEN, EQOptionInt) },
-      Law("Foldable Laws: toList turn items into a list") { FF.`toList turn items into a list`(GEN, EQ) },
-      Law("Foldable Laws: fold returns combination of all items") { FF.`fold should combine all items`(GEN, EQ) },
-      Law("Foldable Laws: combineAll is an alias for fold") { FF.`combineAll consistent with fold (alias)`(GEN, EQ) },
-      Law("Foldable Laws: reduceLeftToOption combines all items into an optional value") { FF.`reduceLeftToOption returns Option value`(GENB, EQOptionInt) },
-      Law("Foldable Laws: reduceRightToOption combines all items into an optional value") { FF.`reduceRightToOption returns Option value`(GENB, EQOptionInt) },
-      Law("Foldable Laws: reduceLeftOption consistent with reduceLeftToOption") { FF.`reduceLeftOption returns Option value`(GEN, EQOptionInt) },
-      Law("Foldable Laws: reduceRightOption consistent with reduceRightToOption") { FF.`reduceRightOption returns Option value`(GEN, EQOptionInt) }
-    )
+    val EQListKInt = ListK.eq(Int.eq())
+
+    return laws(Option.traverseFilter(), Option.applicative(), Option.genK(), Option.eqK()) +
```

Comment:
hmm. Actually I was wrong. We need the overload that you had in the previous commit. But only for the `orEmpty` test. Do not use this to test anything that works over arbitrary `Applicative` instances (like `traverse_` and `sequence_`). This way we have `orEmpty` tested whenever our type is both `Foldable` and `Applicative` and we test everything else in the other cases. Oversight on my part sorry :pray:

Reply:
Sure, no problem!! I'll change it back for `orEmpty` api only.. but I have a question about that too, for some typeclasses like `MapK`, `SetK` and `SortedMapK` there wasn't the possibility to use `Applicative`. Those can be skipped.. or shall I use `Id`/`Option.applicative()` ?

Reply:
Those cannot use `orEmpty` anyway so that doesn't need to be tested in that case. They can use `traverse_` and `sequence_` because that works with any applicative (just use `Id` or `Option`) and thus it needs to be tested

Reply:
Something like this ?

Reply:
Almost, no need to test `traverse_` and `sequence_` separately. But this is a good start.